File: examples/node/quickstart.mjs
import { Jio } from "@jio/sdk";

const jio = new Jio({
  endpoint: process.env.JIO_ENDPOINT,
  apiKey: process.env.JIO_API_KEY,
});
const vm = await jio.create();
console.log(`created ${vm.id}`);

try {
  await vm.writeFile(
    "/home/jio/message.txt",
    Buffer.from("hello from Node.js\n"),
  );
  const result = await vm.exec(
    "python -c \"from pathlib import Path; " +
      "print(Path('/home/jio/message.txt').read_text().upper(), end='')\"",
  );
  if (!result.success) {
    throw new Error(result.stderr.toString("utf8"));
  }
  process.stdout.write(result.stdout);

  const contents = await vm.readFile("/home/jio/message.txt");
  if (contents.toString("utf8") !== "hello from Node.js\n") {
    throw new Error("Jio returned different file contents");
  }
} finally {
  await vm.destroy();
}

File: examples/python/quickstart.py
import os

from jio import Jio


jio = Jio(
    api_key=os.environ["JIO_API_KEY"],
)
vm = jio.create()
print(f"created {vm.id}")

try:
    vm.write_file("/home/jio/message.txt", b"hello from Python\n")
    result = vm.exec(
        "tr '[:lower:]' '[:upper:]' < /home/jio/message.txt"
    )
    result.raise_for_status()
    print(result.stdout_text, end="")

    contents = vm.read_file("/home/jio/message.txt")
    assert contents == b"hello from Python\n"
finally:
    vm.destroy()
